tree/tree/src/TTree.cxx

```diff
@@ -8220,6 +8220,41 @@ void TTree::MarkEventCluster()
     ++fNClusterRange;
 }
 
+/// Estimate a reasonable cache size based on the cluster sizes.
+/// This method is meant to be called if all other heuristics fail (see GetCacheAutoSize()).
+Long64_t TTree::EstimateCacheSize()
```

Comment:
`GetCacheAutoSize` intent/purpose is to estimate the [default] cache size :), so this name is a bit confusing.
Consider renaming this function `EstimateClusterSize` and move the final calculation (number of clusters to cache size) into the `GetCacheAutoSize`.

Reply:
done (i called the new method `GetMedianClusterSize`, split the PR in two commits)